samcli/cli/command.py

```diff
@@ -104,6 +96,117 @@ def _set_commands(package_names):
 
         return commands
 
+    def format_options(self, ctx, formatter):
+        # Re-order options so that they come after the commands.
+        self.format_commands(ctx, formatter)
+        opts = [RowDefinition(text="\n")]
+        for param in self.get_params(ctx):
+            row = param.get_help_record(ctx)
+            if row is not None:
+                term, help_text = row
+                opts.append(RowDefinition(name=term, text=help_text))
+
+        if opts:
+            with formatter.indented_section(name="Options", extra_indents=1):
+                formatter.write_rd(opts)
+
+        with formatter.indented_section(name="Examples", extra_indents=1):
+            formatter.write_rd(
+                [
+                    RowDefinition(
+                        text="\n",
+                    ),
+                    RowDefinition(
+                        name="Get Started:",
+                        text=click.style(f"${ctx.command_path} init"),
+                        extra_row_modifiers=[ShowcaseRowModifier()],
+                    ),
+                ],
+            )
+
+    def format_commands(self, ctx, formatter):
+        with formatter.section("Commands"):
+            with formatter.section("Create an App"):
```

Comment:
nit: I am in between whether each section here should be a method/class of its own and handle how it writes the commands section. Since this won't be touched too much I am OK with this approach as well. 🤔

Reply:
I thought about this as well, but left it this way as its apparent to the user and it won't be touched as often, we can refactor as we fit later.